### PR TITLE
Externalize session-service properties

### PR DESCRIPTION
**Pull request by wtritch**
Externalize configuration properties:
1. Remove properties from docker-container startup script
2. Add bootstrap.yml properties for connecting to a spring cloud config service
3. Add required maven dependencies
4. Add runtime application.yml with default properties and a local-development profile
5. Add test application.yml with default testing properties to allow running unit/integration tests with a local database

### CODE REVIEW

**Review by millerts — APPROVED** [review]

File: service/src/main/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  undertow:
+    buffer-size: 16384
+    buffers-per-region: 20
+    io-threads: 64
+    worker-threads: 512
+    direct-buffers: true
+
+session-service:
+  #Assume we're also running a tds-exam-service in "local-development" mode
+  exam-url: http://localhost:8081
```
Comment (jtreuting) [inline service/src/main/resources/application.yml:28]: I feel like I've seen `examUrl` used before in other PRs for other services?

**Review by jtreuting — COMMENTED** [review]
Just one quick question about consistency across config files and services.  not a blocker though

File: service/src/main/resources/application.yml (hunk 1 above)
Comment (wtritch) [reply service/src/main/resources/application.yml:28]: I haven't seen it in the associated ".properties" files on the wiki.  I can do a quick scan, however.

Comment (jtreuting) [reply service/src/main/resources/application.yml:28]: I meant I think you've used examUrl in a PR that had come through.  Not a big deal and maybe I'm missing remembering :)

Comment (wtritch) [reply service/src/main/resources/application.yml:28]: It doesn't *look* like any of the other services communicate directly with the exam service, but I'm going off of a source code scan.  I'd feel better if somebody with deep knowledge of the system interactions checked me =)

Comment (jtreuting) [reply service/src/main/resources/application.yml:28]: None of the other micro services do.  TDS_Student does (not student service but the main student application).  Is say just ignore my comment cause I'm probably thinking of something else.

Nothing to see here ... Move along

**Pull request merged**